yrs/src/undo.rs

```diff
@@ -479,29 +477,35 @@ where
     where
         O: Into<Origin>,
     {
-        let inner = self.inner();
+        let inner = Arc::get_mut(&mut self.state).unwrap();
         inner.options.tracked_origins.remove(&origin.into());
     }
 
     /// Clears all [StackItem]s stored within current UndoManager, effectively resetting its state.
-    pub fn clear(&mut self) -> Result<(), TransactionAcqError> {
-        let inner = self.inner();
-        let mut txn = inner.doc.try_transact_mut()?;
+    ///
+    /// # Deadlocks
+    ///
+    /// In order to perform its function, this method must guarantee that underlying document store
+    /// is not being modified by another running `TransactionMut`. It does so by acquiring
+    /// a read-only transaction itself. If transaction couldn't be acquired (because another
+    /// read-write transaction is in progress), it will hold current thread until, acquisition is
+    /// available.
+    pub fn clear(&mut self) {
```

Comment:
It looks like there is no `try_clear()`, as you did for other methods?

Reply:
Yes, I'll need to add that in subsequent release. As of now, it will block current thread but only if there's a pending read-write transaction (since clear only requires a readonly access.